lumicks/pylake/detail/image.py

```diff
@@ -114,8 +114,11 @@ def line_timestamps_image(time_stamps, infowave, pixels_per_line):
     pixels_per_line : int
         The number of pixels on the fast axis of the scan.
     """
-    min_stamps = reconstruct_image(time_stamps, infowave, shape=(pixels_per_line,), reduce=np.min)
-    return min_stamps.astype(np.int64)[:, 0]
+    infowave, valid_idx = discard_zeros(infowave)
+    time_stamps = time_stamps[valid_idx]
+    pixel_start_idx = np.flatnonzero(infowave == InfowaveCode.pixel_boundary)
+    pixel_start_idx = np.concatenate(([0], pixel_start_idx + 1))
+    return time_stamps[pixel_start_idx[0:-1:pixels_per_line]]
```

Comment:
I think that your fix from the `dtype_experimental` branch is much better. It fixes the actual problem. This just creates yet another way to reconstruct data based on the infowave. I'm also a bit puzzled why you chose a very different reconstruction method than `reconstruct_image()`. I'd opt for our fix from the other branch that fixes the core issue in `reconstruct_image()`.

Reply:
I agree that the other fix definitely has long term preference. The reason I opted for the pointfix now is because fixing `reconstruct_image` requires finding a workaround for the potential overflow issues when taking the mean of timestamps and making sure that none of the downstream algorithms are affected by a change from floating point to integer datatypes.

I went with this form because it effectively reverses #124 where this flaw was introduced. After the "good" `dtype` fix, this function should be changed back to `reconstruct_image`. In fact, after that fix, it should be possible to get rid of this function entirely and use the `reduce_timestamps` factory that's already in there for this (with argument `reduce=np.min`). That said, that means that for safety, we should test this not just at the image level, but also add the `Kymo` level. I have just added an extra test for this.

If you prefer, I can change it to be similar in structure as `reconstruct_image`.

Reply:
Thanks for the explanation. Makes sense to leave it as is for this PR.